### ISSUE TITLE
Local and proxied catalogues should include all supported search methods in the metadata

### ISSUE THREAD

**Issue by smulube** [HyperCatIoT/hc2-server#1]
This reference server supports all the HyperCat 2.0 search methods doesn't it, so shouldn't it both include this information in the local example catalogue's metadata, but also insert this metadata to proxied catalogues?

From my very quick scanning about of the code, I could see where to add it for the example local catalogue, but was less sure about where this might be inserted for proxied catalogues.

**Comment by tobyjaffey** [comment]
Yes, I agree. When proxying, the server should be adding the extra catalogue metadata to declare support for search mechanisms. If people are going to use hc2-server as a real proxy, we'll need this. However, our coming client code will know it's dealing with hc2-server and make the right assumptions.
